src/js/item.ts

```diff
@@ -46,35 +52,64 @@ export function extractItems(
                 order_elem,
                 context + ';extractItems:strategy:' + i,
             );
+
             if (items.length) {
                 return items;
             }
         } catch (ex) {
             console.error('strategy' + i.toString() + ' ' + ex);
         }
     }
+
     return [];
 }
 
+
+// TODO not the best place for this method, should rearrange and possibly change the callsite where this is used
+export function getCategoriesForProduct(scheduler: IRequestScheduler, productUrl: string): Promise<string> {
+    return scheduler.scheduleToPromise<string>(
+        productUrl,
+        // the result of this transformation is cached; clear cache when debugging & changing this line
+        // the scheduler cache uses localstorage which has a very limited storage size, so do not store full page results
+        (evt) => {
+            const productPage = util.parseStringToDOM(evt.target.responseText).documentElement;
+
+            try {
+                return util.findSingleNodeValue('//*[@id="wayfinding-breadcrumbs_feature_div"]/ul', productPage, '').textContent.
+                // remove all duplicate spaces and newlines. This creates a reasonably formatted category breadcrumb.
+                replace(/\n|\r|[ ]{2,}/g, "")
+            } catch (ex) {
+                // if the breadcrumb doesn't exist, the category is highlighted bold on a submenu bar, let's extract that
+                return util.findSingleNodeValue('//*[@id="nav-subnav"]/a[contains(@class, "nav-b")]', productPage, '').textContent.trim()
+            }
+        },
+        '00000',
```

Comment:
I cannot remember if this is the right priority to avoid deadlock - how sure are you (if you're not, then I'm happy to dig around and figure it out)?

Reply:
I'm not sure on this!